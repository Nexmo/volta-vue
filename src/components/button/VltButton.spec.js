import { mount } from '@vue/test-utils';
import { VltButton } from '..';

describe('vlt-button', () => {
  test('renders correctly', () => {
    const wrapper = mount(VltButton);
    expect(wrapper.element).toMatchSnapshot();
  });

  test.each([
    ['icon', 'bin'],
    ['primary', true],
    ['id', 'test-button'],
    ['disabled', true],
    ['spinner', true],
    ['spinner-white', true],
  ])('prop %s is set to %s', (prop, value) => {
    const wrapper = mount(VltButton, {
      context: {
        props: {
          [prop]: value,
        },
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
