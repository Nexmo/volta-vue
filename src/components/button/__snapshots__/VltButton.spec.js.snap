// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vlt-button prop disabled is set to true 1`] = `
<button
  class="Vlt-btn Vlt-btn--no-focus Vlt-btn--icon Vlt-btn--disabled Vlt-btn--tertiary"
>
  <!---->
   
</button>
`;

exports[`vlt-button prop icon is set to bin 1`] = `
<button
  class="Vlt-btn Vlt-btn--no-focus Vlt-btn--icon Vlt-btn--tertiary"
>
  <svg
    class="Vlt-icon"
  >
    <use
      xlink:href="#V_Vlt-icon-bin"
    />
  </svg>
   
</button>
`;

exports[`vlt-button prop id is set to test-button 1`] = `
<button
  class="Vlt-btn Vlt-btn--no-focus Vlt-btn--icon Vlt-btn--tertiary"
  id="test-button"
>
  <!---->
   
</button>
`;

exports[`vlt-button prop primary is set to true 1`] = `
<button
  class="Vlt-btn Vlt-btn--no-focus Vlt-btn--icon Vlt-btn--primary"
>
  <!---->
   
</button>
`;

exports[`vlt-button renders correctly 1`] = `
<button
  class="Vlt-btn Vlt-btn--no-focus Vlt-btn--icon Vlt-btn--tertiary"
>
  <!---->
   
</button>
`;
